### PR TITLE
Support cluster name association

### PR DESCRIPTION
**Pull request by tgraf**
Required for #4727

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4933)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/node/node.go [hunk 1]
```diff
@@ -61,6 +69,16 @@ type Node struct {
 	cluster *clusterConfiguation
 }
 
+// Fullname returns the node's full name including the cluster name if the
+// cluster name has been specified
+func (n *Node) Fullname() string {
```
Comment (aanm) [inline pkg/node/node.go]: I don't understand this. How can you know if the cluster name has been specified or not if you are assigning a default value ("default") to the CLI flag?

File: pkg/node/node.go [hunk 2]
```diff
@@ -16,26 +16,34 @@ package node
...
 type Identity struct {
-	Name string
+	Name    string
+	Cluster string
 }
 
 // String returns the string representation on NodeIdentity.
 func (nn Identity) String() string {
-	return nn.Name
+	return path.Join(nn.Cluster, nn.Name)
```
Comment (aanm) [inline pkg/node/node.go:36]: Is `String()` being used for any for of identification of this node in the kvstore? If yes, will this break the ability to do a downgrade?

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/node/node.go (hunk 1 above)
Comment (tgraf) [reply pkg/node/node.go]: We can't. I can clarify the comment to say: specified a value other than the default value.

File: pkg/node/node.go (hunk 2 above)
Comment (tgraf) [reply pkg/node/node.go:36]: It was used internally to identify but we don't have any APIs that depend on it.

File: pkg/node/node.go (hunk 1 above)
Comment (tgraf) [reply pkg/node/node.go]: fixed

**Comment by tgraf** [comment]
test-me-please

File: pkg/node/node.go (hunk 1 above)
Comment (aanm) [reply pkg/node/node.go]: There was nothing wrong with the documentation, what I was trying to say was if the user uses the cluster name "default" then he will not see in the controllers outputs for example in `cilium status`.

**Review by ianvernon — APPROVED** [review]
Approving for docs / API changes.

File: pkg/node/node.go (hunk 1 above)
Comment (tgraf) [reply pkg/node/node.go]: Yes, that's intentional. Do you see a problem with that? You would not expect to run multiple clusters with the name default.

**Review by aanm — APPROVED** [review]

**Pull request merged**